src/core/authorization_request/verification/did.rs

```diff
@@ -40,6 +40,14 @@ pub async fn verify_with_resolver(
         bail!("request was signed with unsupported algorithm: {alg}")
     }
 
+    // This bypass is for unencoded JWT requests, but we will need to change this later
+    // so that trust is preserved when receiving unencoded requests
+    // NOTE: This requires that `Algorithm::None` is permitted in the wallet metadata
+    // Otherwise, this function will error in the previous assertion.
+    if alg.contains("none") {
+        return Ok(());
+    }
+
```

Comment:
According to the spec, I think we should check whether the `kid` exists before we check for `alg` is set to `none`.

See: https://datatracker.ietf.org/doc/rfc9101/ 

> The authorization server MUST validate the signature of the JWS-
   signed [RFC7515] Request Object.  If a "kid" Header Parameter is
   present, the key identified MUST be the key used and MUST be a key
   associated with the client.  The signature MUST be validated using a
   key associated with the client and the algorithm specified in the
   "alg" Header Parameter.  Algorithm verification MUST be performed, as
   specified in Sections 3.1 and 3.2 of [RFC8725].

>   If the key is not associated with the client or if signature
   validation fails, the authorization server MUST return an
   "invalid_request_object" error to the client in response to the
   authorization request.

Reply:
I think [this requirement](https://openid.net/specs/openid-4-verifiable-presentations-1_0-20.html#section-5.7-6.2.1) from openid4vp supercedes this requirement:

> Since DID Document may include multiple public keys, a particular public key used to sign the request in question **MUST be identified by the kid in the JOSE Header**.

Reply:
I agree this requirement takes precedence. The question I have now, though, is that it seems the missing `kid` value in the header is in nonconformance with the oid4vp specification.

We are currently bypassing the `kid` check IFF the wallet metadata supports a `None` algorithm for signing.

This preserves the explicit check `kid`, but allows the wallet metadata to control bypassing this assertion.

@cobward What suggested changes do you think we should make here?

@Joey-Silberman anything you want to add?

Reply:
If time doesn't permit making the update elsewhere (in mobile-sdk), then I don't think we should block merging this in. Just ensure that we clarify the issue with the relevant parties and fix this at the earliest opportunity.